scipy/sparse/sparsetools/bsr.h

```diff
@@ -796,13 +805,14 @@ void bsr_matvecs(const I n_brow,
     const npy_intp Y_bs = (npy_intp)n_vecs*R; //Yx blocksize
     const npy_intp X_bs = (npy_intp)C*n_vecs; //Xx blocksize
 
+    scal(n_brow * R * n_vecs, beta, Yx);
```

Comment:
It depends, but that's a good point. I can move it down into the loops for `csr_matvecs` and both `bsr` routines.  

With `csc` I don't see any way to do it other than by scaling `y` first.  The question of loading `y` twice is moot for `csc` anyway since it accesses `y` randomly.  

In the general case `dia_matvec` already makes `n_diags` passes through `y` so I'm not very worried about adding one more.  `n_diags <= 3` could be special cased.  I'm thinking primarily here of diagonal matrices where doing two loops through `y` vs. one loop might be much worse.

Scaling first is also required for `coo` but that doesn't really matter.

Will update.